pkg/snmp/traps/forwarder.go

```diff
@@ -44,13 +46,16 @@ func (tf *TrapForwarder) Stop() {
 }
 
 func (tf *TrapForwarder) run() {
+	flushTicker := time.NewTicker(10 * time.Second).C
 	for {
 		select {
 		case <-tf.stopChan:
 			log.Info("Stopped TrapForwarder")
 			return
 		case packet := <-tf.trapsIn:
 			tf.sendTrap(packet)
+		case <-flushTicker:
+			tf.sender.Commit() // Commit metrics
```

Comment:
A `if tf.sender != nil` probably doesn't hurt here.

Reply:
@remeh I'm not sure if `if tf.sender != nil` will bring value here since:

1/ we already check for sender not being returned here https://github.com/DataDog/datadog-agent/blob/cf8024835612a38028e57834b665f9758436ba22/pkg/snmp/traps/server.go#L36-L39

2/ and `sender.Count/EventPlatformEvent/etc`  are already used in multiple places in NDM Traps code, even if we add a protection here, it will fail on those other places if sender is `nil` (but shouldn't happen thanks to 1/).